### ISSUE TITLE
Datasets location

### ISSUE THREAD

**Issue by gestefane** [vaseline555/Federated-Learning-in-PyTorch#7]
Please, can you provide more information about datasets location as exact name for files to download, website, etc? Thank you!

**Comment by Ofir408** [comment]
I think you can use the following code for download Mnist:

```
import torch
from torchvision import datasets

torch.utils.data.DataLoader(
    datasets.MNIST('.', download=True))
```

**Comment by Ofir408** [comment]
> Please, can you provide more information about datasets location as exact name for files to download, website, etc? Thank you!


I think you can use the following code for download Mnist:

```
import torch
from torchvision import datasets

torch.utils.data.DataLoader(
    datasets.MNIST('.', download=True))
```

**Comment by vaseline555** [comment]
Sorry for super late reply.
In current implementation, I did not consider the case of using custom dataset.
But it is not that difficult; you can retrieve the public dataset (e.g., datasets in `torchivision`) you want as @Ofir408 kindly replied, or you may implement custom `torch.utils.data.Dataset` class instance and just feed it to the `DataLoader`.

FYI, if you have interest in ready-made FL simulation framework, you may find [another repository of mine](https://github.com/vaseline555/Federated-Learning-with-Flower) useful. It is implemented upon [Flower](https://flower.dev) federated learning framework.
Or, if you want to use public FL benchmark (e.g., LEAF benchmark) implemented in PyTorch, please refer to [the other repository of mine](https://github.com/vaseline555/superfed). It is an implementation of my paper on the personalized federated learning, [Connecting Low-Loss Subspace for Personalized Federated Learning](https://arxiv.org/abs/2109.07628) published in ACM 28th SIGKDD conference.

Thank you.
